src/Support/OperationExtensions/ResponderExtension.php

```diff
@@ -0,0 +1,149 @@
+<?php
+
+namespace Dedoc\Scramble\Support\OperationExtensions;
+
+use Dedoc\Scramble\Extensions\OperationExtension;
+use Dedoc\Scramble\Support\Generator\Combined\AnyOf;
+use Dedoc\Scramble\Support\Generator\Operation;
+use Dedoc\Scramble\Support\Generator\Response;
+use Dedoc\Scramble\Support\Generator\Schema;
+use Dedoc\Scramble\Support\Generator\Types as OpenApiTypes;
+use Dedoc\Scramble\Support\RouteInfo;
+use Dedoc\Scramble\Support\Type\Generic;
+use Illuminate\Support\Collection;
+use PhpParser\Node\Stmt\ClassMethod;
+use PhpParser\Node\Stmt\Return_;
+
+class ResponderExtension extends OperationExtension
+{
+    public function handle(Operation $operation, RouteInfo $routeInfo)
+    {
+        $definition = $routeInfo->methodNode();
+
+        if (!$definition instanceof ClassMethod) {
+            return;
+        }
+
+        $statements = $definition->getStmts();
+
+        $returnStatements = array_filter($statements, function ($statement) {
+            return $statement instanceof Return_
+                && $statement->expr instanceof \PhpParser\Node\Expr\MethodCall
+                && $this->usesResponderClass($statement->expr);
+        });
+
+        $responses = collect($returnStatements)
+            ->map(function ($returnStatement) {
+                $statusCode = $this->getStatusCode($returnStatement->expr);
+                $transformer = $this->getTransformer($returnStatement->expr);
+
+                if (!$transformer) {
+                    return null;
+                }
+
+                $response = $this->openApiTransformer->toResponse(new Generic($transformer));
+                $response->code = $statusCode ?? 200;
+
+                return $response;
+            })
+            ->filter();
+
+        [$responses, $references] = $responses->partition(fn ($r) => $r instanceof Response);
+
+        $responses = $responses
+            ->groupBy('code')
+            ->map(function (Collection $responses, $code) {
+                if (count($responses) === 1) {
+                    return $responses->first();
+                }
+
+                return Response::make((int) $code)
```

Comment:
```suggestion
                return Response::make($code)
```
You don't need this if you type `$code` in the callback params.